apis/config/v1beta1/defaults.go

```diff
@@ -46,10 +48,19 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return nil
 }
 
+func getOperatorNamespace() string {
+	if data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns
+		}
+	}
+	return DefaultNamespace
+}
+
 // SetDefaults_Configuration sets default values for ComponentConfig.
 func SetDefaults_Configuration(cfg *Configuration) {
 	if cfg.Namespace == nil {
-		cfg.Namespace = pointer.String(DefaultNamespace)
+		cfg.Namespace = pointer.String(getOperatorNamespace())
```

Comment:
Change the documentation of the Namespace field to indicate where the default value comes from.